### PR TITLE
[#2518] feat(PyClient): Integrate pytest with Gradle to run python test with CICD

### PR DESCRIPTION
**Pull request by xunliu**
### What changes were proposed in this pull request?

1. Use Gradle python plugin to manage python module
2. Automatic download conda and create python env in `gravitino/.gradle/python` directory
3. Use multiple python version `3.8`, `3.9`, `3.10`,and `3.11` test ITs in Github Action
4. Support `gradlew clean` & `gradlew build` & `gradlew :client:client-python:test -PpythonVersion=3.8`

### Why are the changes needed?

Use Gradle manager and development Python client module.

Fix: #2518 

### Does this PR introduce _any_ user-facing change?

1. Use `pythonVersion` Gradle parameter setting Python version to compile and test Python client, `gradlew :client:client-python:test -PpythonVersion=3.8`
2. Use `skipPythonITs` Gradle parameter to control if skip Python test cases.

### How was this patch tested?

CI Passed.

### CODE REVIEW

**Comment by xunliu** [comment]
hi @shaofengshi  @zhaoyongjie @coolderli Please help me review this PR, Thanks!

**Comment by shaofengshi** [comment]
Hi Xun, I have two comments:

1. is the "python-gradle-miniforge-plugin" a formal gradle plugin? The project seems not under maintained, the last commit is 2 years ago;
2. conda seems more heavy than venv; I'm not sure whether it can switch to venv;

For other parts, it looks good to me

**Comment by xunliu** [comment]
hi @shaofengshi 
> 1. is the "python-gradle-miniforge-plugin" a formal gradle plugin? The project seems not under maintained, the last commit is 2 years ago;

Yes, Because Python Gradle plugin is very few, I didn't found better  than to `python-gradle-miniforge-plugin`.
I think `python-gradle-miniforge-plugin` fulfill our needs. 

> 2. conda seems more heavy than venv; I'm not sure whether it can switch to venv;

I didn't found a Python Gradle plugin would to support venv.

I think we could use it first.

**Comment by shaofengshi** [comment]
> hi @shaofengshi
> 
> > 1. is the "python-gradle-miniforge-plugin" a formal gradle plugin? The project seems not under maintained, the last commit is 2 years ago;
> 
> Yes, Because Python Gradle plugin is very few, I didn't found better than to `python-gradle-miniforge-plugin`. I think `python-gradle-miniforge-plugin` fulfill our needs.
> 
> > 2. conda seems more heavy than venv; I'm not sure whether it can switch to venv;
> 
> I didn't found a Python Gradle plugin would to support venv.
> 
> I think we could use it first.

OK, we can go with this first, and refine later.

**Review by shaofengshi — APPROVED** [review]

File: clients/client-python/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  val runPyTests by registering(VenvTask::class) {
+    dependsOn(pipInstall)
+    venvExec = "pytest"
+    workingDir = projectDir.resolve("tests")
+  }
+
+  compileJava {
+    dependsOn(pipInstall)
```
Comment (coolderli) [inline clients/client-python/build.gradle.kts:43]: should the `compile` and `spotless` depend on `pip install`?  Python doesn't need to compile and can't verify the correctness of the program. If the user only updates the Java code, they also need to download the dependencies of the Python module. How about just making the `test` depend on `pip install`?

**Comment by jerryshao** [comment]
Is this python environment removable, where is it installed?

**Comment by xunliu** [comment]
> Is this python environment removable, where is it installed?这个python环境是可移动的，安装在哪里？

Automatic download `conda` and create `python env` in `gravitino/.gradle/python` directory

File: clients/client-python/build.gradle.kts (hunk 1 above)
Comment (xunliu) [reply clients/client-python/build.gradle.kts:43]: Yes, Currently use Gradle to comple Python module have this limit (need add `compileJava`).
I Create an issue #2770 to track this probleam.

Comment (xunliu) [reply clients/client-python/build.gradle.kts]: hi @coolderli I fixed this problem in this PR. :-)

**Comment by xunliu** [comment]
@shaofengshi @zhaoyongjie @coolderli  Please help me review this PR, Thanks.

File: clients/client-python/setup.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 setup(
     name="gravitino",
     description="project description TBD",
```
Comment (shaofengshi) [inline clients/client-python/setup.py:11]: We can put some description, like "Python lib/client for Gravitino"

File: .github/workflows/python-integration-test.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        with:
+          java-version: ${{ matrix.java-version }}
+          distribution: 'temurin'
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Package Gravitino
+        run: |
+          ./gradlew build -x test -PjdkVersion=${{ matrix.java-version }}
```
Comment (coolderli) [inline .github/workflows/python-integration-test.yml:79]: Can we just build the `clients:client-python` here? The whole build step has already been covered by `build.yml`.

**Comment by coolderli** [comment]
@xunliu Hi, I left two comments. Overall, LGTM. Thanks.

File: .github/workflows/python-integration-test.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            source_changes:
+              - .github/**
+              - api/**
+              - bin/**
+              - catalogs/**
+              - clients/client-java/**
```
[the remark this answers is not in the document]
Comment (xunliu) [reply .github/workflows/python-integration-test.yml]: Thank you for your suggestion.

**Review by shaofengshi — APPROVED** [review]

**Comment by shaofengshi** [comment]
LGTM

**Pull request merged**